### PR TITLE
Update starship configuration

### PR DESCRIPTION
**Pull request by Brettm12345**
I added the `disabled` `symbols` and `styles` options

### CODE REVIEW

File: modules/programs/starship.nix [hunk 1]
```diff
@@ -30,6 +30,34 @@ in {
       description = "The package to use for the starship binary.";
     };
 
+    disabled = mkOption {
```
Comment (rycee) [inline modules/programs/starship.nix]: I don't see what advantage these options offer over setting them in the `settings` option. Are you mainly looking for the improved merge behavior? I think the correct way forward in that case would be to improve the type of `settings`.

Comment (Brettm12345) [reply modules/programs/starship.nix]: Possibly. I just wanted something declarative

Comment (Brettm12345) [reply modules/programs/starship.nix]: I'm still not sure what idiomatic nix is

Comment (rycee) [reply modules/programs/starship.nix]: I made a PR to introduce a more elaborate type for the `settings` option. This new type should allow correct merging of configurations (see the test case in the PR).

**Pull request closed without merging**